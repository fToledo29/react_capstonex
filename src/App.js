import React from 'react';
import './App.css';
import { Menu } from './components/menu/menu';
import { 
	Route,
	BrowserRouter as Router
} from 'react-router-dom';
import Switch from 'react-bootstrap/esm/Switch';
import { About } from './components/about/about';
import { Home } from './components/home/home';
import { SignIn } from './components/authentication/signIn/signIn';
import { library } from '@fortawesome/fontawesome-svg-core';
import { fas } from '@fortawesome/free-solid-svg-icons';
// import { ProductList } from './components/productManagemen/all-products-page/viewProductList/viewProductList';
import AllProductsPage from './components/productManagemen/all-products-page/all-products-page';
library.add(fas);

function App() {
  return (
	<div className="App">
		<Router>
			<Menu></Menu>
			<Switch>
				<Route exact path="/" component={Home} />
				<Route path="/about" component={About} />
				<Route path="/login" component={SignIn} />
				<Route path="/products" component={AllProductsPage} />
				{/* <Route path="/addProduct" component={AddProductPage} />
				<Route path="/product/:productName" component={ProductDetail} /> */}
			</Switch>
		</Router>
	</div>
  );
}

export default App;
