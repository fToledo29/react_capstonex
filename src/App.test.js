import React from 'react';
import 'enzyme-to-json';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {

	let wrapper;
	let props;

	beforeAll(() => {

		props = {
			vistActions: {} ,
			actions: {} ,
			visitData: {} ,
			product: {} ,
			match: {params: {id: 1}, isExact: true, path: "", url: ""},
		};

		wrapper = shallow(<App {...props}/>);

	});

	it('App Renders correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});	