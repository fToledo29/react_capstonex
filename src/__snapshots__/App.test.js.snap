// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Renders correctly 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <Connect(Menu) />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/about"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(SignIn)",
            "type": [Function],
          }
        }
        path="/login"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(withRouter(Register))",
            "type": [Function],
          }
        }
        path="/register"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(AllProductsPage)",
            "type": [Function],
          }
        }
        path="/products"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(AddProductPage)",
            "type": [Function],
          }
        }
        path="/addProduct"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(withRouter(ViewProductPage))",
            "type": [Function],
          }
        }
        path="/viwProduct/:id"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(withRouter(ChartsPage))",
            "type": [Function],
          }
        }
        path="/chart"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(UserDetails)",
            "type": [Function],
          }
        }
        path="/userDetails"
      />
    </Switch>
  </BrowserRouter>
</div>
`;
